### PR TITLE
Asynchronously fetch projects from Wordpress API

### PR DESCRIPTION
**Pull request by Thusnake**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -53,6 +53,61 @@ const mapQuarterIdToName = (id) => {
 
 const showLogo = false;
 
+function ProjectDetails({ project }) {
```
Comment (emiltsokov) [inline src/App.js]: could be extracted to its own file

File: src/App.js [hunk 2]
```diff
@@ -53,6 +53,61 @@ const mapQuarterIdToName = (id) => {
...
+  let [imgUrl, setImgUrl] = React.useState('https://via.placeholder.com/250?text=SpasiSofia');
```
Comment (emiltsokov) [inline src/App.js]: const

File: src/ProjectsProvider.js [hunk 3]
```diff
@@ -26,9 +31,69 @@ export const useProjects = () => {
...
+
+
 export const ProjectsProvider = ({ children }) => {
   let [searchParams] = useSearchParams();
 
+  const [mapTopicIdToTopic, setMapTopicIdToTopic] = React.useState(null);
+  React.useEffect(() =>
+    fetch(`${BASE_URL}/wp-json/wp/v2/project-category/`)
+      .then(res => res.json())
+      .then(data => setMapTopicIdToTopic(new Map(data.map(obj => [obj.id, obj.name]))))
```
Comment (emiltsokov) [inline src/ProjectsProvider.js]: don't we need a catch clause?

File: src/ProjectsProvider.js [hunk 4]
```diff
@@ -26,9 +31,69 @@ export const useProjects = () => {
...
+    , []);
+
+  const [wordpressProjectsRaw, setWordpressProjectsRaw] = React.useState([]);
+  React.useEffect(() => fetch(`${BASE_URL}/wp-json/wp/v2/projects`)
+    .then(res => res.json())
```
Comment (emiltsokov) [inline src/ProjectsProvider.js]: same question as above , don't we need a catch clause ?

File: src/ProjectsProvider.js [hunk 5]
```diff
@@ -26,9 +31,69 @@ export const useProjects = () => {
...
+    .then(data => setWordpressProjectsRaw(data.filter(el => el.acf && el.acf.map))),
+    []);
+
+  const [wordpressProjects, setWordpressProjects] = React.useState([]);
+  React.useEffect(() =>
+    setWordpressProjects(wordpressProjectsRaw.map(el => Object({
```
Comment (emiltsokov) [inline src/ProjectsProvider.js]: issue: 
you probably need this 
React.useEffect(async() =>
    setWordpressProjects(
await Promise.all(wordpressProjectsRaw.map(async el =>

check 
https://stackoverflow.com/questions/33438158/best-way-to-call-an-asynchronous-function-within-map

also first prepare/normalise data and after that set it in the state

File: src/App.js (hunk 1 above)
Comment (georgibogdanov) [reply src/App.js]: Extracted in separate file

File: src/App.js (hunk 2 above)
Comment (georgibogdanov) [reply src/App.js]: done

File: src/ProjectsProvider.js (hunk 3 above)
Comment (georgibogdanov) [reply src/ProjectsProvider.js]: Added

File: src/ProjectsProvider.js (hunk 4 above)
Comment (georgibogdanov) [reply src/ProjectsProvider.js]: Added catch

File: src/ProjectsProvider.js (hunk 5 above)
Comment (georgibogdanov) [reply src/ProjectsProvider.js]: Used Promise.all as suggested and explained. I think we are good to do it this way as technically it happens only once.

**Review by emiltsokov — APPROVED** [review]

**Pull request merged**